cmd/testServer/main.go

```diff
@@ -0,0 +1,178 @@
+	"io"
+	"log"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"google.golang.org/grpc"
+)
+
+const BUCKETNAME = "secrets"
+const ENDPOINT = "localhost:9000"
+const ACCESSKEYID = "masoud"
+const SECRETACCESSKEY = "Strong#Pass#2022"
+const ISUSESSL = false
+
+func initBucket(ctx context.Context, bucketName string, client *minio.Client) error {
+	exists, err := client.BucketExists(ctx, bucketName)
+	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+
+	if !exists {
+		err = client.MakeBucket(ctx, bucketName, minio.MakeBucketOptions{})
+		if err != nil {
+			return fmt.Errorf("%w", err)
+		}
+	}
+	return nil
+}
+
+func initMinioRepo(endpoint, accessKeyID, secretAccessKey string, isUseSSL bool) (*minio.Client, error) {
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(accessKeyID, secretAccessKey, ""),
+		Secure: isUseSSL,
+	})
+	if err != nil {
+		fmt.Printf("Error init client minio: %s", err.Error())
+
+		return nil, fmt.Errorf("%w", err)
+	}
+
+	return minioClient, nil
+}
+
+func main() {
+
+	conn, err := grpc.Dial("localhost:50051", grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+
+	// client := pb.NewGophKeeperServiceClient(conn)
+
+	// stream, err := client.BlockStore(context.Background())
+	// if err != nil {
+	// 	log.Fatalf("Error opening stream: %v", err)
+	// }
+
+	// go func(stream pb.GophKeeperService_BlockStoreClient) {
+	// 	for {
+	// 		select {
+	// 		case <-stream.Context().Done():
+	// 			fmt.Println("stream canceled")
+
+	// 			return
+	// 		default:
+	// 			connState := conn.GetState()
+	// 			if connState != connectivity.Ready {
+	// 				fmt.Println("stream canceled or not ready")
+
+	// 				return
+	// 			}
+	// 			fmt.Println("all ok")
+	// 			fmt.Println(conn.GetState())
+	// 			time.Sleep(1 * time.Second)
+	// 		}
+	// 	}
+	// }(stream)
+
+	go func() {
+		for {
+			fmt.Println(conn.GetState())
+			time.Sleep(1 * time.Second)
+		}
+	}()
+
+	// // Отправка нескольких BlockStoreRequest сообщений
+	// for i := 0; i < 10000; i++ {
+	// 	req := &pb.BlockStoreRequest{
+	// 		Guid:     "guid-" + string(i),
+	// 		IsFinish: false,
+	// 	}
+	// 	if err := stream.Send(req); err != nil {
+	// 		log.Fatalf("Failed to send request: %v", err)
+	// 	}
+	// 	time.Sleep(2 * time.Second) // Задержка между отправкой сообщений
+	// }
+
+	// // Завершение стрима с is_finish = true
+	// if err := stream.Send(&pb.BlockStoreRequest{Guid: "final-guid", IsFinish: true}); err != nil {
+	// 	log.Fatalf("Failed to send final request: %v", err)
+	// }
+
+	// // Получение ответа
+	// resp, err := stream.CloseAndRecv()
+	// if err != nil {
+	// 	log.Fatalf("Failed to receive response: %v", err)
+	// }
+	// log.Printf("Response: %s", resp.Status)
+
+	// Для демонстрации: ожидание, чтобы увидеть, как "keepalive" работает
+	time.Sleep(3000 * time.Second)
+}
+
+func main_minio() {
```

Comment:
Please rename it

Reply:
testServer/ is a development and debugging package. does not carry a functional load for services